### ISSUE TITLE
Some loop types don't decompile correctly, causing code duplication in the output

### ISSUE THREAD

**Issue by krzys-h** [krzys-h/UndertaleModTool#94]
Some loop types like do...while and especially repeat break the loop decompiler causing the code to be duplicated in the output

Also see #85

**Comment by colinator27** [comment]
I was writing the code to compile repeat statements (referencing the compiler to make sure I was doing it right), and here's what I found that may help in identifying them. You may already know the structure, but I'm just clarifying that some parts of it are always the same.

Repeat statements begin with the expression representing loop count, of course (`repeat(<expression>)`). After that is a possible conversion to Int32.

Following that are these exact instructions, always (except for the branching target, obviously):
```
dup.i 0
push.i 0
cmp.i.i LTE
bt <cleanup_label>
```
(by the way, for some reason this loop uses `push.i` instead of `pushi.e`? that's quite strange...)

After that, the actual body of the repeat statement is written. For this example, we'll call the label that goes to the beginning of it `body_label`.

At the end of the body, the "decrement" section runs, using these exact instructions:
```
decrement_label: push.i 1
sub.i.i
dup.i 0
conv.i.b
bt <body_label>
```
(`decrement_label` is unused unless the `continue` statement is used explicitly)

Finally, at the end is the cleanup:
```
cleanup_label: popz.i
```
(`cleanup_label` can also be jumped to if a `break` statement is used explicitly)

**Comment by krzys-h** [comment]
So it's mostly what I expected.

tl;dr: it uses a tempvar on the stack for the iteration counter

**Comment by colinator27** [comment]
I really wonder whose idea it was to make it decrement instead of increment though...

**Comment by krzys-h** [comment]
It's shorter so more efficient, I guess. Even without looking at the assembly `for(i=0; i<n; i++)` is way more instructions than `while(n--)` (since it doesn't need original value of n nor the current iteration value and can compare with 0 without explicit cmp instruction)

**Comment by colinator27** [comment]
Oh, yeah, I didn't even think of that!

**Comment by colinator27** [comment]
Fixed in #162
